check_tags.go

```diff
@@ -0,0 +1,62 @@
+package trapcheck
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/circonus-labs/go-apiclient"
+)
+
+func (tc *TrapCheck) UpdateCheckTags(ctx context.Context, tags []string) (*apiclient.CheckBundle, error) {
+	if tc.checkBundle == nil {
+		return nil, fmt.Errorf("invalid state, check bundle is nil")
+	}
+	if len(tags) == 0 {
+		return nil, nil
+	}
+
+	update := false
+	for _, tag := range tags {
+		if tag == "" {
+			continue
+		}
+		found := false
+		tagParts := strings.SplitN(tag, ":", 2)
+		for j, ctag := range tc.checkBundle.Tags {
+			if tag == ctag {
+				found = true
+				break
+			}
+
+			ctagParts := strings.SplitN(ctag, ":", 2)
+			if len(tagParts) == len(ctagParts) {
+				if tagParts[0] == ctagParts[0] {
+					if tagParts[1] != ctagParts[1] {
```

Comment:
What about tags that are not in the `category:value` format? Such as tags like `category:`, `:value` or just `tag`? I'm not sure if any of these tag formats are allowed for check tags, but the API docs describe allowing tags like `tag` and considering them "uncategorized". In any case, I'm not sure if we can always assume tagParts[1] or ctagParts[1] will exist, and might need to check the index before accessing it.

Reply:
I believe the logic will still work, but another set of eyes on it would be great.

```
[]string{"foo", "bar"}  // "foo:bar"
[]string{"foo", ""} // "foo:"
[]string{"", "bar"} // ":bar"
[]string{"baz"} // "baz"
```